File: src/main/java/de/fraunhofer/aisec/analysis/structures/TypestateMode.java

package de.fraunhofer.aisec.analysis.structures;

public enum TypestateMode {

	/**
	 * Non-deterministic finite automaton. Intraprocedural, not alias-aware.
	 */
	NFA,

	/**
	 * Weighted Pushdown System. Interprocedural, alias-aware, context-aware.
	 */
	WPDS

}

File: codyze-v3/codyze-backends/cpg/src/main/kotlin/TypestateMode.kt
package de.fraunhofer.aisec.codyze_backends.cpg

/** Enum containing all supported typestate modes. */
enum class TypestateMode {
    /** Non-deterministic finite automaton. Intraprocedural, not alias-aware. */
    DFA,

    /** Weighted Pushdown System. Interprocedural, alias-aware, context-aware. */
    WPDS
}
